app/src/components/core/metaschema/metaSchema.ts

```diff
@@ -16,102 +16,146 @@ module app.core.metaschema {
+                    var name = definitionsRefs[i]['$ref'].substring(("#/definitions/").length);
+                    var value = json['definitions'][name];
                     if (value.hasOwnProperty('type')) {
                         if (value['type'] === 'object') {
-                            var definition:Definition = new Definition(name, Metaschema.cleanupDataschema(value), Metaschema.retrieveAcceptedElements(value, rootElements));
+                            var definition:Definition = new Definition(name, Metaschema.cleanupDataschema(value, json), Metaschema.retrieveAcceptedElements(value, definitionsNames));
 
                             // now continue with all child elements that the current element can accept
-                            _.forEach(Metaschema.retrieveChildElements(value, rootElements), (child:Definition) => {
+                            _.forEach(Metaschema.retrieveChildElements(value, definitionsNames, json), (child:Definition) => {
                                 definitions.push(child);
                             });
 
                             // add everything to the result
                             definitions.push(definition);
                             _.forEach(definition.getTypeLabels(), (type:string) => {
-                                if (rootElements.indexOf(type) === -1) {
-                                    rootElements.push(type);
+                                if (definitionsNames.indexOf(type) === -1) {
+                                    definitionsNames.push(type);
                                 }
                             });
                         }
                     }
-                });
+                }
             }
             return new Metaschema(definitions);
         }
 
-        private static retrieveAcceptedElements(value:{}, rootElements:string[]):string[] {
+        private static retrieveAcceptedElements(value:{}, definitionsNames:string[]):string[] {
             var result:string[] = [];
-            if (value.hasOwnProperty('properties') && value['properties'].hasOwnProperty('elements') && value['properties']['elements'].hasOwnProperty('items')) {
-                var items = value['properties']['elements']['items'];
-
-                if (items.hasOwnProperty('$ref') && items['$ref'] === '#') {
-                    result = rootElements;
-                } else {
-                    // a new object is declared, so get its label
-                    if (items.hasOwnProperty('properties') && items['properties'].hasOwnProperty('type') && items['properties']['type'].hasOwnProperty('enum')) {
-                        result = items['properties']['type']['enum'];
-                    }
+            var items = null;
+
+            if (value['allOf']) {
+                var index;
+                for (var i = 0; i < value['allOf'].length; i++) {
+                    if (value['allOf'][i]['properties']) index = i;
```

Comment:
We are looking for the index at which there's an object with a `properties` keyword, right? Then we should be able to break out of the loop here.